pkg/common/url.go

```diff
@@ -27,3 +28,19 @@ func URLForExposedServiceInPlugin(serviceName string, plugin *greenhousev1alpha1
 		subdomain, plugin.GetNamespace(), DNSDomain,
 	)
 }
+
+// SplitHost splits an exposed service host into its name, namespace and cluster parts
+// The pattern shall be $service--$cluster--$namespace
+// TODO, need to fix: We know that information on the namespace and even the cluster might be lost when the host was hashed due to it's length.
+// Currently only cluster is critical, need to fix this.
+func SplitHost(host string) (name, cluster, namespace string, err error) {
+	parts := strings.SplitN(host, ".", 2)
+	if len(parts) < 2 {
+		return "", "", "", fmt.Errorf("invalid host: %s", host)
+	}
+	parts = strings.SplitN(parts[0], "--", 3)
+	if len(parts) < 3 {
+		return "", "", "", fmt.Errorf("invalid host: %s", host)
+	}
+	return parts[0], parts[1], parts[2], nil
+}
```

Comment:
Just a Suggestion - 

Would you prefer to return a struct instead and a verbose error on where it failed which can logged? This way we may not have to deal with multiple string and order?

Example:

```go
type ServiceProxyHost struct {
	Name      string
	Cluster   string
	Namespace string
}
```

```go
func SplitHost(host string) (ServiceProxyHost, error) {
	// Split the host into the name part and domain part
	parts := strings.SplitN(host, ".", 2)
	if len(parts) < 2 {
		return ServiceHost{}, fmt.Errorf("invalid host: missing domain: %s", host)
	}

	// Further split the first part (before the first dot) into service, cluster, and namespace
	parts = strings.SplitN(parts[0], "--", 3)
	if len(parts) < 3 {
		return ServiceHost{}, fmt.Errorf("invalid host: expected format $service--$cluster--$namespace: %s", host)
	}

	return ServiceProxyHost{
		Name:      nameParts[0],
		Cluster:   nameParts[1],
		Namespace: nameParts[2],
	}, nil
}
```

Reply:
Yes that makes a lot of sense. I would like to create an issue to properly tackle a refactoring.
This now just moves the relevant functions into one file for easier reference and hot fixes the faulty order.